gems/aws-sdk-bedrock/lib/aws-sdk-bedrock/plugins/bearer_authorization.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+module Aws::Bedrock
+  module Plugins
+    # @api private
+    class BearerAuthorization < Seahorse::Client::Plugin
+      def after_initialize(client)
+        return unless (token = ENV['AWS_BEARER_TOKEN_BEDROCK'])
+
+        client.config.token_provider ||= Aws::StaticTokenProvider.new(token)
+      end
+
+      class Handler < Seahorse::Client::Handler
+        def call(context)
+          # This also sets the preferred auth scheme even if the code token has precedence.
+          context[:auth_scheme] = { 'name' => 'bearer' } if ENV['AWS_BEARER_TOKEN_BEDROCK']
+          with_metric { @handler.call(context) }
+        end
+
+        private
+
+        def with_metric(&block)
+          if ENV['AWS_BEARER_TOKEN_BEDROCK']
```

Comment:
Possible nit - this should also check that the token wasn't overridden in code

Reply:
I'm clarifying this with Aaron.